### PR TITLE
[MODORDSTOR-259] - Added acquisition method API

### PR DESCRIPTION
**Pull request by Dmitriy-Butramyou**
## Purpose
Add acquisition method API

## Approach
Creader migration
Added system value in acquisition_method table
Added API

## Learning

@folio-org/acquisitions

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Dmitriy-Butramyou** [comment]
Working of  the migration verified by postman
image

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -126,6 +126,37 @@
...
       ]
     },
+    {
+      "id": "orders-storage.acquisition-methods",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/orders-storage/acquisition-methods",
+          "permissionsRequired": ["orders-storage.acquisition-methods.get"]
```
Comment (AndreiMPV) [inline descriptors/ModuleDescriptor-template.json:136]: Should be "orders-storage.acquisition-methods.collection.get"

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -612,6 +643,43 @@
         "orders-storage.po-lines.item.delete"
       ]
     },
+    {
+      "permissionName" : "orders-storage.acquisition-methods.get",
```
Comment (AndreiMPV) [inline descriptors/ModuleDescriptor-template.json:647]: Same "orders-storage.acquisition-methods.collection.get"

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -126,6 +126,37 @@
...
+        },
+        {
+          "methods": ["GET"],
+          "pathPattern": "/orders-storage/acquisition-methods/{id}",
+          "permissionsRequired": ["orders-storage.acquisition-method.item.get"]
+        },
+        {
+          "methods": ["DELETE"],
+          "pathPattern": "/orders-storage/acquisition-methods/{id}",
+          "permissionsRequired": ["orders-storage.acquisition-method.delete"]
```
Comment (AndreiMPV) [inline descriptors/ModuleDescriptor-template.json:156]: Should be "orders-storage.acquisition-method.item.delete"

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -612,6 +643,43 @@
...
+      "displayName" : "acquisition-method-item get",
+      "description" : "Fetch a acquisition-method"
+    },
+    {
+      "permissionName" : "orders-storage.acquisition-method.item.put",
+      "displayName" : "acquisition-method put",
+      "description" : "Update a purchase acquisition method"
+    },
+    {
+      "permissionName" : "orders-storage.acquisition-method.delete",
```
Comment (AndreiMPV) [inline descriptors/ModuleDescriptor-template.json:667]: Should be "orders-storage.acquisition-method.item.delete"

**Review by AndreiMPV — CHANGES_REQUESTED** [review]

**Review by AndreiMPV — APPROVED** [review]

**Review by mogrim — APPROVED** [review]

**Pull request merged**